lib/dictionary_processing.py

```diff
@@ -183,17 +181,21 @@ def get_paths_inside_metamodel(service, service_dict, deprecate_rest=False, repl
                 path = service_dict[service].operations[operation_id].metadata[request].elements['path'].string_value
                 path_list.add(path)
 
-                if deprecate_rest and not is_in_rest_navigation:
+                # If already found in navigation, no need to check for request mapping
+                if not is_in_rest_navigation:
                     is_rest_api_existing = check_for_request_mapping_replacement(service_dict[service], operation_id)
 
-                if deprecate_rest and not is_rest_api_existing and not is_rest_navigation_checked:
+                if not is_rest_api_existing and not is_rest_navigation_checked:
                     is_rest_api_existing = check_for_rest_navigation_replacement(service_url, rest_navigation_handler)
                     # Add all operations and methods to replacements if it is apparent in rest_navigation
                     is_in_rest_navigation = is_rest_api_existing
                     is_rest_navigation_checked = True
 
                 if is_rest_api_existing:
                     add_replacement_path(service, operation_id, request.lower(), path, replacement_dict)
+                    # If a newer version of the api is included - remove it unless deprecation is on
+                    if not deprecate_rest and not blacklist_utils.is_blacklisted_for_rest(service):
+                        path_list.remove(path)
```

Comment:
Is it terribly harder to not add it in the first place instead of removing later? removing always smells to me, at the very least it mutates the conteiner unnecessarily.

Reply:
Removed the `remove()`